### ISSUE TITLE
[LionButton] @click event listener is called when button has `disabled`  and click is trigger programatically using JS `.click()` method

### ISSUE THREAD

**Issue by jhuesos** [ing-bank/lion#2330]
## Expected behavior

When a lion-button has the attribute `disabled`, the JavaScript `.click()`  method should not trigger the @click listener(s)

## Actual Behavior

Clicking with mouse does respect the disabled attribute, but when calling with the .click() it does not.

https://stackblitz.com/edit/vite-zh1vmm?file=main.js

## Additional context

Which component and version of it are you using.

**Comment by aghArdeshir** [comment]
I looked into this for a few hours:
- It is hard to reproduce it in the automated tests for some reason. But it is easily reproducible in the context of a normal consumer app
- Simply putting this line in the `LionButton.js` seems to solve this issue, and I don't know why:

![image](https://github.com/user-attachments/assets/95a6129e-c74f-419a-8418-e8a1d049b8f7)

And its hard to vlaidate this fix, because it is hard to put this problem in an automated test

**Comment by aghArdeshir** [comment]
I found about the reason that prevented me from writing a test for this bug: https://github.com/ing-bank/lion/issues/2351

After fixing https://github.com/ing-bank/lion/issues/2351, I should be unblocked for working on this bug with disabled lion-buttons handling clicks

**Comment by aghArdeshir** [comment]
This PR: https://github.com/ing-bank/lion/pull/2354 fixes this issue

**Comment by denilsonsa** [comment]
Hi! When @jhuesos showed me the bug in `lion-button`, I decided to check if the native elements would also behave the same way. So I created a small testcase showing [the behavior of native HTML elements when being clicked while disabled](https://codepen.io/denilsonsa/pen/zYVZRVX). I'm sharing the link here because I think it is relevant, and it can serve as a baseline to build a similar testcase for lion input fields.
